app/models/concerns/slug_bug.rb

```diff
@@ -22,14 +19,15 @@ def set_slug
                 else
                   id
                 end
+    self.slug_for_upgrade = self.slug
   end
 
   private
 
     # Cribbed from https://gitlab.com/notch8/louisville-hyku/-/blob/main/app/models/custom_slugs/slug_behavior.rb#L14
     def remove_index_and_reindex
       # rubocop:disable Rails/Blank
-      return unless slug.present?
+      return unless (slug.present? || slug_for_upgrade.present?)
```

Comment:
> * existing works will have a slug property
> * new works will have slug and slug_for_upgrade

I'm going to go a little overkill on my explanation because this is a delicate dance and I want to make sure I understand and to give everyone the opportunity to share and correct my understanding.

We have three places to consider: Fedora, Ruby, SOLR.  Once we deploy, in Ruby `slug_for_upgrade` will be a copy/alias of `slug`.  At the point of deploy, no Fedora nor SOLR document will have the `slug_for_upgrade` property.  When someone saves an existing Fedora object then both the SOLR and Fedora documents will have values for the `slug_for_upgrade` property.

I'm uncertain what happens if we simply reindex an existing Fedora object.  Will the SOLR document have the `slug_for_upgrade`?

Similarly when someone creates a new Fedora object it will set both the `slug` and `slug_for_upgrade` attributes on the Fedora and SOLR document.

> once the migration happens, we want to use slug_for_upgrade? and so the check for either property here ensures that this method works now and in the future?

The goal is to slowly move from `slug (predicate DC:alternate)` to `slug_for_upgrade (predicate slug)` to `slug (predicate slug)`.  This first step is a "prepare the way" step.